docs/taskman-design.md

```diff
@@ -0,0 +1,379 @@
+
+This implementation provides a clean, modular solution for that problem by exposing a gRPC-based API to start and manage processes on a remote server. Users can interact with this API through a command-line interface (CLI), making the system easy to use for both people and automation workflows (e.g., CI/CD pipelines, job queues, developer tools).
+
+## Details
+
+#### Required Approvers
+
+```
+# Required Approvers (two required)
+* Engineering:  @tigrato || @timothyb89 || @russjones || @zmb3 || @rosstimothy || @r0mant
+```
+
+### UX
+
+The server is started by running the taskman-server binary, which launches the gRPC service responsible for task lifecycle management. The listening address can be configured using the --server-address flag, defaulting to localhost:50051 if not specified.
+
+In a production setting, additional flags could be introduced to configure --ca-key, --server-key, --server-cert, --log-level, and other operational parameters.
+```
+$ taskman-server --help
+Usage:
+  taskman-server [--server-address] [--help]
+
+Description:
+  This service manages task lifecycle (start, stop, status) and streams output to clients over a secure mTLS connection.
+
+Options:
+  --server-address <host:port>
+      The address the gRPC server will listen on (e.g., localhost:50051). Defaults to localhost:50051 if not set.
+  --help
+      Display help information for the server command.
+Example:
+  $ taskman-server --server-address localhost:12345
+```
+
+Users will interact with the exposed gRPC server APIs through commands provided by the `taskman` command line tool. This tool will utilize a gRPC client to communicate with the gRPC server. The user must pass a `--user-id` flag that allows the user to select which client they are. The CLI will pick from a hardcoded set of certificates in the application that will be part of the gRPC requests in order to perform authentication. The following CLI commands will be provided in the taskman client:
+
+#### start
+The start command starts a new task given the passed in command and associates the task with the provided user-id. If successful the generated task-id will be returned.
+
+```
+$ taskman start --help
+Usage:
+  taskman start --user-id <user-id> [--server-address <host:port>] [--help] -- <command> [args...]
+
+Description:
+  Start a new task by executing the specified command. The command may be passed as a quoted string (for shell interpretation), or as separate arguments if no shell features are needed. The --user-id flag is required to identify the client initiating the request.
+
+Arguments:
+  <command> [args...]
+        The command to execute, followed by any optional arguments. The command can be passed as:
+        - A quoted string for shell interpretation (e.g., "ls /myFolder | grep foo")
+        - A command with space-separated arguments (e.g., ls /myFolder)
+
+        The binary can be a full path or must exist in the system's PATH.
+        Example: "ls"
```

Comment:
I think this is still a bit ambiguous, since some rule is needed to decide whether or not to use shell interpretation (and which shell, etc).

I'd argue the simpler and better solution might be to let the user's own shell do the parsing, and only support the 2nd case. If the user wants to run a shell command, they can just use e.g. `sh -c 'command with spaces'`

Reply:
I like this approach better, I will update.